bpfd/src/utils.rs

```diff
@@ -23,21 +23,34 @@ pub(crate) fn get_ifindex(iface: &str) -> Result<u32, BpfdError> {
     }
 }
 
-pub(crate) async fn set_map_permissions(directory: &str) {
+pub(crate) async fn set_dir_permissions(directory: &str, mode: u32) {
     // Determine if User Group exists, if not, do nothing
     if get_group_by_name(USRGRP_BPFD).is_some() {
         // Iterate through the files in the provided directory
         for file in fs::read_dir(directory).unwrap() {
-            // Set the permissions on the map such that members of the "bpfd" group can read/write
+            // Set the permissions on the file based on input
             if (tokio::fs::set_permissions(
-                file.unwrap().path(),
-                fs::Permissions::from_mode(0o0660),
+                file.as_ref().unwrap().path(),
+                fs::Permissions::from_mode(mode),
             )
             .await)
                 .is_err()
             {
-                info!("Unable to set permissions on map. Continuing");
+                info!(
+                    "Unable to set permissions on file {}. Continuing",
+                    file.unwrap().path().display()
+                );
             }
         }
     }
 }
+
+pub(crate) async fn set_file_permissions(path: &str, mode: u32) {
+    // Determine if User Group exists, if not, do nothing
+    if get_group_by_name(USRGRP_BPFD).is_some() {
+        // Set the permissions on the file based on input
+        if (tokio::fs::set_permissions(path, fs::Permissions::from_mode(mode)).await).is_err() {
+            info!("Unable to set permissions on file {}. Continuing", path);
```

Comment:
Same as the previous comment on info vs warn, this feels more like a warning to me.